### PR TITLE
feat: extract commands out to platform packages

### PR DESCRIPTION
**Pull request by grabbou**
Summary:
---------

Follow up to #290, we also move out commands.

Side note: In the future, we might want to move `errors` to tools too in order to have better error reporting. Not important for now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "glob": "^7.1.3",
     "jest": "^24.6.0",
     "lerna": "^3.13.1",
+    "metro-memory-fs": "^0.53.1",
```
Comment (thymikee) [inline package.json]: Please move it to `devDependencies`, it's only used for testing, no need to feed it to the users

File: packages/cli/package.json [hunk 2]
```diff
@@ -57,10 +52,7 @@
     "semver": "^5.0.3",
     "serve-static": "^1.13.1",
     "shell-quote": "1.6.1",
-    "slash": "^2.0.0",
-    "ws": "^1.1.0",
-    "xcode": "^2.0.0",
-    "xmldoc": "^0.4.0"
+    "ws": "^1.1.0"
```
Comment (thymikee) [inline packages/cli/package.json]: would be good to update it, latest is v6 :o

File: packages/cli/src/commands/link/unlink.js [hunk 3]
```diff
@@ -137,7 +136,9 @@ function unlink(args: Array<string>, ctx: ConfigT) {
     })
     .catch(err => {
       logger.error(
-        `It seems something went wrong while unlinking. Error:\n${err.message}`,
+        `It seems something went wrong while unlinking. CLIError:\n${
+          err.message
+        }`,
```
Comment (thymikee) [inline packages/cli/src/commands/link/unlink.js]: How about actually throwing `CLIError` here?

Comment (grabbou) [reply packages/cli/src/commands/link/unlink.js]: yeah, could be.

File: packages/cli/package.json (hunk 2 above)
Comment (grabbou) [reply packages/cli/package.json]: Yeah, but this is proxy stuff, so I am a bit worried of doing it here.

**Pull request merged**